packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -836,14 +839,26 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         private _storage?: IDocumentStorageService,
     ) {
         super();
-
+        this.metadata = metadata ?? this.formMetadata();
         this.baseSummaryMessage = metadata?.message;
         /**
           * gcFeature in metadata is introduced with v1 in the metadata blob. Forced to 0/disallowed before that.
           * For existing documents, we get this value from the metadata blob.
           * For new documents, we get this value based on the gcAllowed flag in runtimeOptions.
           */
         const prevSummaryGCVersion = existing ? getGCVersion(metadata) : undefined;
+
+        /**
+         * If container exists, then get these values from metadata.
+         * If there is no metadata or metadata does not have these values, it means this is an old document,
+         * so these values will be undefined. If container doesn't exist, we initialize these values
+         */
+        this.metadata.createContainerRuntimeVersion = existing ? metadata?.createContainerRuntimeVersion ?? undefined
+            : pkgVersion;
+        this.metadata.createContainerTimeStamp = existing ? metadata?.createContainerTimeStamp ?? undefined
+            : performance.now();
```

Comment:
```suggestion
        this.metadata.createContainerRuntimeVersion = existing
            ? metadata?.createContainerRuntimeVersion ?? undefined
            : pkgVersion;
        this.metadata.createContainerTimeStamp = existing
            ? metadata?.createContainerTimeStamp ?? undefined
            : performance.now();
```

Style Nit: If you split a ternary operator across lines, it's easier to read like this.

Reply:
Also, I don't think you need the `??`.  That maps only `null` and `undefined` to the default case given. Since you're always giving `undefined` as the default case, and I'm assuming `null` is not an expected/supported value here we want to code for, it's equivalent to just the LHS (e.g. `metadata?.createContainerRuntimeVersion`).  See https://mariusschulz.com/blog/nullish-coalescing-the-operator-in-typescript

Reply:
Ok reading this code again... can you just do this?

```
if (!existing) {
  this.metadata.foo = 0;  // Do the right values for each prop ofc
```

Because if `metadata` is defined, then `this.metadata === metadata`, right?

Reply:
Err see my comment about about the initialization of `this.metadata`, I think you need to rework this object slightly